userspace/libscap/scap.h

```diff
@@ -836,12 +836,17 @@ int32_t scap_get_stats(scap_t* handle, OUT scap_stats* stats);
 /*!
   \brief Returns the set of ppm_sc whose events have EF_MODIFIES_STATE flag or whose syscall have UF_NEVER_DROP flag.
 */
-int scap_get_modifies_state_ppm_sc(uint32_t * ppm_sc_array);
+int scap_get_modifies_state_ppm_sc(OUT uint32_t ppm_sc_array[PPM_SC_MAX]);
```

Comment:
I don't quite like this signature, but i understand the self-documenting case.
I'd prefer to just add a line of documentation and keep the pointer there though; but no strong opinion on this one.

Reply:
(same for tracepoints below)

Reply:
Not sure probably i prefer somthing like this since it is more explicit 
```c
int scap_get_modifies_state_ppm_sc(OUT uint32_t ppm_sc_array[PPM_SC_MAX])
```
@Molter73 WDYT?

Reply:
Honestly, having array sizes specified in function definitions always looks a bit weird IMO, but if we can be this explicit about what the function is receiving, I'd prefer this method, as long as we don't care about compatibility with C89, (33 years and this is still a big deal).

Reply:
if we care about compatibility with C89 i think is time to delete the repo :rofl:

Reply:
Uhm as far as i know, @gnosek told me that they are actually using an old gcc without full c99 compatibility, so...

Reply:
c99 or c89 ? :laughing:

Reply:
The gcc they're using is not compatible with c99, so c89!
But i think i am wrong because we are using Designated initializers in all libscap engines now, and that is a c99 feature.

Reply:
> if we care about compatibility with C89 i think is time to delete the repo 🤣

Don't dismiss it so easily, there are plenty of systems out there with compilers that only support C89, I worked on such a system only 4 years ago.